openedx/features/discounts/utils.py

```diff
@@ -9,7 +9,7 @@
 from .applicability import can_receive_discount, discount_percentage
 
 
-def format_strikeout_price(user, course, base_price=None):
+def format_strikeout_price(user, course, base_price=None, already_checked_discount=False):
```

Comment:
this took me a little bit to figure out. What do you think about changing the param to `check_for_discount=True`, and then passing false if you've already checked?

Reply:
sure, will update